### PR TITLE
DAOS-2336 mgmt: Add MS pool directory

### PR DESCRIPTION
**Pull request by liw**
This patch adds a pool directory (i.e., a pool KVS) in MS to track all
the pools in the system. The directory maps each pool UUID to a pool
record, which includes the pool service replica ranks.

The Go changes make sure MS is created and started even when the system
is using PMIx, so that the pool directory is available in both PMIx and
PMIx-less modes.

The new mgmt dRPC SetUp and the server init state ensure that
daos_io_server only sets up the server modules and prints the "DAOS I/O
server" message when daos_server has finished starting MS. This helps
tests that wait for the "DAOS I/O server" message before invoking pool
create commands.

As daos_io_server will require mgmt to start, this patch adds mgmt to
the module list used by rdb_test_runner.py.

Signed-off-by: Li Wei <wei.g.li@intel.com>

### CODE REVIEW

**Comment by liw** [comment]
Rebased to the latest master to resolve dRPC conflicts.

**Comment by liw** [comment]
@tanabarr, would you be able to review this patch? I'd like to try and see if I can get this landed before going on vacation tomorrow.

**Comment by johannlombardi** [comment]
Tom, could you please rebase this patch since LiWei is on vacation? It seems to be conflicting with your pool destroy patch. Thanks in advance.

File: src/control/server/iosrv.go [hunk 1]
```diff
@@ -49,6 +50,30 @@ import (
...
+
+// pmixRank returns the PMIx rank. If PMIx-less, it returns nilRank.
+func pmixRank() rank {
+	s, ok := os.LookupEnv("PMIX_RANK")
+	if !ok {
+		return nilRank
+	}
+	r, err := strconv.ParseUint(s, 0, 32)
+	if err != nil {
+		panic(fmt.Sprintf("PMIX_RANK=%s: %v", s, err))
```
Comment (mjmac) [inline src/control/server/iosrv.go]: This should not panic. Malformed user input is not an exceptional situation. Please change the return signature to include an error value or else just return nil rank.

Comment (liw) [reply src/control/server/iosrv.go]: @mjmac, PMIX_RANK is set by orterun internally, not from users. All PMIX_RANK accesses are hacks that will go away once we fully switch to PMIx-less mode.

Comment (mjmac) [reply src/control/server/iosrv.go]: I understand -- but I don't think it's a good practice to ever use panic in production code, even if it's supposed to go away eventually. Why not just return the nil rank?

Comment (liw) [reply src/control/server/iosrv.go]: I still think this panic is fine, but I've changed the function to return (rank, error).

**Pull request merged**